src/analyzer.rs

```diff
@@ -255,6 +255,8 @@ fn extract_container(
                 }
                 if let Some(dict) = dict_key {
                     format!("{}<String, {}>", cfg.map.name(), dict)
+                } else if is_ref(value) {
+                    "ObjectReference".into()
```

Comment:
you're technically only finding it as a key in an object now, right?
if it's inside an array we would not catch it?

Reply:
Yeah, probably not, I’ll add a test case for it. Currently this is only used for resolving these two refs in Cluster API spec: https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/cluster_types.go#L62-L67

Reply:
> Added, tests should pass.

thanks a lot. looks great. going to merge it as is for now. i do think there are extra things we __can__ do here, but not sure the best way forward myself. i do like that kopium is very transparent (with the command run in the generated header) so i would ideally keep that to make any migrations for users easy. not sure how a config with work with that, but open to suggestions.

it's also possible that the flag we've made here should work for [more types of objectreferences](https://docs.rs/k8s-openapi/latest/k8s_openapi/?search=ObjectRef) in a later version. i see there are [tons of variants used in the crd registry repo](https://github.com/search?q=repo%3Ametio%2Fkube-custom-resources-rs+objectref+language%3AYAML&type=code&l=YAML). one step at a time though, this is great.

Reply:
As to an idea of how usage of CLI flags can be combined with config - I actually did this in https://github.com/crust-gather/crust-gather/, it was the main motivation to continue this project. Using clap `derive` with arg value_parser and serde allowed me to achieve 3 simultaneous implementations by adding only one, or at least I felt this way while writing this
1. good documentation for CLI arguments and the structure fields
2. new CLI flag for added functionality
3. new serde value to automatically parse with --config, inheriting work from previous 2

I believe something like this can be done with `kopium`, since it is also CLI driven and using `clap`. And because every config value is essentially a CLI flag, here it will be compatible with transparent annotation on top of generated file, even if a sort of --config flag is used. It will be converted to a list of CLI flags I think.

Config: https://github.com/crust-gather/crust-gather/blob/main/src/cli.rs#L117
Regular CLI option for the same stuct: https://github.com/crust-gather/crust-gather/blob/main/src/cli.rs#L50